### ISSUE TITLE
When using `isolation.enabled` and `ingress.enabled`, `vcluster connect` does not work

### ISSUE THREAD

**Issue by felipecrs** [loft-sh/vcluster#873]
### What happened?

It gets stuck at `Waiting for vcluster to come up...`.

### What did you expect to happen?

To work. If I disable `isolation.networkPolicy.enabled`, then it works.

### How can we reproduce it (as minimally and precisely as possible)?

I think the issue is self-explanatory, but if you need more details please let me know.

### Anything else we need to know?

I believe this is probably because the necessary rules aren't being added to `networkPolicy` in such situation `ingress.enabled`.

### Host cluster Kubernetes version

<details>

```console
$ kubectl version
Client Version: v1.26.0
Kustomize Version: v4.5.7
Server Version: v1.21.14
```

</details>

### Host cluster Kubernetes distribution

<details>

```
Vanilla
```

</details>

### vlcuster version

<details>

```console
$ vcluster --version
0.13.0
```

</details>

### Vcluster Kubernetes distribution(k3s(default)), k8s, k0s)

<details>

```
k3s
```

</details>

### OS and Arch

<details>

```
OS: Ubuntu 18.04
Arch: x64
```
</details>

**Comment by FabianKramm** [comment]
Hey @felipecrs ! Thanks for creating this issue and sorry for the late response, yes this seems to be caused by the network policy, does it work when disabling the network policy?

**Comment by felipecrs** [comment]
Yes, this is caused by network policy created by vcluster. I think it lacks a rule for the ingress communication.

**Comment by FabianKramm** [comment]
@felipecrs we have an option for this `isolation.networkPolicy.outgoingConnections.ipBlock.expect`, where you can specify the ip ranges to not block. Otherwise I'm not sure if there would be a generic solution as this probably depends on the ingress controller / load balancer used.

**Comment by felipecrs** [comment]
@FabianKramm, I believe the helm template function `getHostByName` could be used to get the IP address of the ingress host and thus, automatically add it to the list of allowed addresses in the network policy.
